src/1_api.js

```diff
@@ -51,12 +51,32 @@ Server.prototype.serializeObject = function(obj, prefix) {
  * @param {Object.<string, *>} data
  */
 Server.prototype.getUrl = function(resource, data) {
-	var k, v, err;
-	var url = resource.destination + resource.endpoint;
+	var k, v, err,
```

Comment:
to be fair - in javascript all var declarations are hoisted to the function scope, there is no block scope - so it doesn't really matter. just a question of keeping code as readable as possible